app/main.py

```diff
@@ -1,25 +1,76 @@
-from abc import ABC
-
-
-class IntegerRange:
-    pass
-
-
-class Visitor:
-   pass
-
-
-class SlideLimitationValidator(ABC):
-    pass
-
-
-class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
-
-
-class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
-
-
-class Slide:
-    pass
+from __future__ import annotations
+from typing import Type
+from abc import ABC
+
+
+class IntegerRange:
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+        self,
+        owner: Type[SlideLimitationValidator],
+        name: str
+    ) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(
+        self,
+        instance: SlideLimitationValidator,
+        owner: Type[SlideLimitationValidator],

```

Comment:
You could define `Any` in the type annotation for `owner` and `instance` to not limit potential places where `IntegerRange` class can be used